# Description: Use ubuntu_gtk_set_use_overlay_scrollbar and
# ubuntu_gtk_get_use_overlay_scrollbar so that other programs can say they
# don't want to use the overlay scrollbars. This workaround some crash in
# Qt apps for instance.
# Author: Albert Astals Cid <albert.astals@canonical.com>
# Bug: http://launchpad.net/bugs/1005677

=== modified file 'os/os-scrollbar.c'
--- a/os/os-scrollbar.c	2012-08-29 16:09:23 +0000
+++ b/os/os-scrollbar.c	2012-09-17 14:07:20 +0000
@@ -3161,7 +3161,11 @@
 static gboolean
 use_overlay_scrollbar (void)
 {
+#ifdef USE_GTK3
   return scrollbar_mode != SCROLLBAR_MODE_NORMAL;
+#else
+  return scrollbar_mode != SCROLLBAR_MODE_NORMAL && ubuntu_gtk_get_use_overlay_scrollbar ();
+#endif
 }
 
 static void
@@ -4231,6 +4235,10 @@
                                    "}\n", -1, NULL);
 #endif
 
+#ifndef USE_GTK3
+  ubuntu_gtk_set_use_overlay_scrollbar (TRUE);
+#endif
+
   /* Load custom overlay scrollbar style. */
   if (use_overlay_scrollbar ())
     custom_style_load ();

